pkg/jobs/job_scheduler.go

```diff
@@ -252,8 +254,20 @@ func (s *jobScheduler) executeSchedules(
 			continue
 		}
 
+		sp, err := txn.CreateSavepoint(ctx)
```

Comment:
@knz or @nvanbenschoten can you just double-check we're using the Savepoint API correctly?